src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php

```diff
@@ -83,5 +84,24 @@ public function process(ContainerBuilder $container)
                 $loader->load('services.yml');
             }
         }
+
+        //@todo: POC proposal. If accepted it probably needs to be moved to dedicated pass, but priority is important
+        //       (it should probably be after this ModulesPass. e.g. using same in ContainerInjectionPass doesn't work)
+        // we find all definitions in order to search all decorated services
+        // @todo: find a way to optimize and find all decorated services or all controllers (cannot find decorated controllers by tags)
+        $allDefinitions = $container->getDefinitions();
+        foreach ($allDefinitions as $definition) {
+            $decoratedService = $definition->getDecoratedService();
+            // if service is decorator
+            if ($decoratedService && $container->hasDefinition($decoratedService[0])) {
+                $decoratedServiceDefinition = $container->getDefinition($decoratedService[0]);
+                // and if service is controller
+                if ($decoratedServiceDefinition->hasTag('controller.service_arguments')) {
+                    // then we add controller.service_arguments tag to the decorator and remove it from the decorated class
+                    $definition->addTag('controller.service_arguments');
+                    $decoratedServiceDefinition->clearTag('controller.service_arguments');
```

Comment:
Why do we need to remove the tag on the original controller?

Reply:
to make it work :smile: (as I said if the tag is in both places then we get error). When controller is decorated it only works when the tag is explicitly declared in the decorator, but must not exist in the decorated class for some reason :sweat: 
I can't find why this happens, maybe the problem is elsewhere :man_shrugging:  :(

Reply:
@matks maybe we don't need that "controller.service_arguments" anymore? You mentioned autowiring, but how can I check that? I tried using autowiring in the decorated controller but i get various random errors that doesn't depend on these changes. Maybe you have a real example of autowiring usage that you mentioned?

Reply:
@zuk3975 I checked my PR https://github.com/PrestaShop/PrestaShop/pull/18811 and I found the reason why I added this tag

https://github.com/PrestaShop/PrestaShop/pull/18811#discussion_r414660147

This tag enable autowiring arguments for controller action methods. It allows controllers to do this:
```
public function createAction(Doctrine $doctrine, $request)
{
     $doctrine->flush();
}
```

It's a little nice-to-have feature I enabled in https://github.com/PrestaShop/PrestaShop/pull/18845